typing/typecore.ml

```diff
@@ -4098,8 +4096,10 @@ and type_expect_
       with Exit ->
         let arg = match sarg with
         | None -> None
-        | Some sarg -> Some (type_exp env argument_mode sarg,
-                        register_allocation expected_mode)
+        | Some sarg ->
+            Some ( type_exp env argument_mode sarg
+                 , register_allocation expected_mode
+                 )
```

Comment:
No spaces after `(`, no `,` to start a line and no dangling `)` please:
```ocaml
            Some (type_exp env argument_mode sarg,
                  register_allocation expected_mode)
```
or:
```ocaml
            Some
              (type_exp env argument_mode sarg,
               register_allocation expected_mode)
```
are acceptable, but this:
```ocaml
            let arg = type_exp env argument_mode sarg in
            let alloc_mode = register_allocation expected_mode in
            Some (arg, alloc_mode)
```
is superior.